kahuna/public/js/util/storage.js

```diff
@@ -0,0 +1,24 @@
+import angular from 'angular';
+
+export const storage  = angular.module('util.storage', []);
+
+storage.factory('storage', ['$window', function($window) {
+    function setItem(key, val) {
+        const storeVal = typeof val === 'string' ? val : JSON.stringify(val);
+        $window.localStorage.setItem(key, storeVal);
+    }
+
+    function getItem(key) {
+        let val = $window.localStorage.getItem(key);
+        try {
+            val = JSON.parse(val);
+        } catch (_) {}
```

Comment:
Not being able to store a string does seems like more of a bug to me, as string is a valid JS type.
Say for instance I was storing a name? Would I have to:
```JS
const n = storage.getItem('name').name;
```
?

We could log, but then if you were expecting a string, you'd get the log every time.

Ah type-unsafety - saying that we could have:
```JS
function getItem(key, isString) {}
```?